### ISSUE TITLE
Question: commitMoveSelectionSet & applyMove

### ISSUE THREAD

**Issue by Merudo** [RPTools/maptool#702]
**Describe your question**

I'm trying to fix #699 but I'm quite confused by the code in [commitMoveSelectionSet](https://github.com/RPTools/maptool/blob/675d5925562185504556e985fb0d1680ea674d6c/src/main/java/net/rptools/maptool/client/ui/zone/ZoneRenderer.java#L439).

The issue is that for each token, the [following differences](https://github.com/RPTools/maptool/blob/675d5925562185504556e985fb0d1680ea674d6c/src/main/java/net/rptools/maptool/client/ui/zone/ZoneRenderer.java#L500) are taken:

```java
        int cellOffX = originPoint.x - tokenCell.x;
        int cellOffY = originPoint.y - tokenCell.y;
```
The weird part is that if the keyToken is not snap-to-grid but the iterated token is, the offsets are a difference of pixels units minus cell units. That seems like an useless difference to take!

These weird offsets are then fed into [applyMove](https://github.com/RPTools/maptool/blob/675d5925562185504556e985fb0d1680ea674d6c/src/main/java/net/rptools/maptool/model/Token.java#L1018):

```java
token.applyMove(
            set, path, set.getOffsetX(), set.getOffsetY(), keyToken, cellOffX, cellOffY);
```

which are then handled to [derive](https://github.com/RPTools/maptool/blob/675d5925562185504556e985fb0d1680ea674d6c/src/main/java/net/rptools/maptool/model/Path.java#L80).

Honestly I'm quite confused by what the code is doing with `cellOffX` and `cellOffY`, does anyone have an idea?

**Comment by Azhrei** [comment]
Yes, I remember seeing that code.  There was some discussion about how it _should_ work, but without agreement, IIRC.

What should it do?  It seems to me that the “leader” (the token clicked on to perform the drag) should be moved by pixels since it’s not STG.  The others should have their coordinates converted to pixels, moved the same amount, then converted back to cells.

If the leader is STG but other selected tokens are not, the same thing:  those that are STG are converted to pixels and all movement is done that way, then converted back to cells for those that are STG.  In this case, though, movement will be in chunks of cell size since the leader is moved by cell-size amounts.

I’m not sure what these conversions back and forth would do to placement on a hex grid, though.

**Comment by JamzTheMan** [comment]
I know Lee did a bunch of work on that STG/non-STG movement, especially around if you moved multiple tokens with a mix.  Problem is the whole movement code is across several classes and can be hard to follow.

Also, some of it was sadly not 100% done I think. So if you see something odd, it may actually just be wrong.

I agree with @Azhrei on that's probably how it should work.  It's a PITA to test too because you need to test STG, NSTG, Combo, and that's across grid/hex/iso/no grid maps, so a lot of testing...

**Comment by Phergus** [comment]
@Azhrei 's description of the expected movement matches how I would expect it to work.

**Comment by Merudo** [comment]
Thank you! You answered my question.
